x/posts/keeper/keeper.go

```diff
@@ -82,15 +99,24 @@ func (k Keeper) GetPostChildrenIDs(ctx sdk.Context, postID types.PostID) types.P
 	return postIDs
 }
 
-// GetPosts returns the list of all the posts that are stored into the current state.
+// GetPosts returns the list of all the posts that are stored into the current state
+//sorted by their incremental ID.
 func (k Keeper) GetPosts(ctx sdk.Context) (posts types.Posts) {
 	posts = types.Posts{}
 	k.IteratePosts(ctx, func(_ int64, post types.Post) (stop bool) {
 		posts = append(posts, post)
 		return false
 	})
 
-	return posts
+	store := ctx.KVStore(k.StoreKey)
+	postsSorted := make(types.Posts, len(posts))
+	for _, post := range posts {
+		var index sdk.Int
+		k.Cdc.MustUnmarshalBinaryBare(store.Get(types.PostIndexedIDStoreKey(post.PostID)), &index)
+		postsSorted[index.Int64()-1] = post
+	}
```

Comment:
While this implementation surely works, I don't think that it's the best. One problem that I see with it is that other places that will use `IteratePosts` will have to copy/paste this piece of code in order to have them ordered as well. 

For this reason, I think that the best way to implement the iteration of posts based on their creation date is to edit the `IteratePosts` method directly. 

Assuming we have two keystore set of key/value pairs defined as: 


| Key | Value |
| :---| :--- |
| `1` | `PostID_a` |
| `2` | `PostID_b` |
| ... | ... |
| `n` | `PostID_z` |

| Key | Value |
| :---| :--- |
| `PostID_a` | Post `a` content |
| `PostID_b` | Post `b` content |
| ... | ... |
| `PostID_z` | Post `z` content |

It should behave as follows: 

1. Iterate over the ordering indexes (`1`, `2`, ... `n`) in *ascending order*. 
2. For each index, get the associated post id. 
3. With the post ID, get the post contents.

Reply:
I tried to switch to this implementation but I'm having problems with the sim tests as the KV store (postIndex, postID) comes in two different orders  inside `ImportExport` test for the way that string sorting is implemented in GO.
Since it's not our interest to keep that store ordered, do you think there's something we could do to exclude it from sim tests? Or it should be ordered as well?  Also some comments are not ordered..I will recheck tomorrow, but I'm not sure if this is the right way to do this

Reply:
@bragaz Your current implementation is based on the following relationships (left side is the key, right side is the value): 

- `PostID -> Index` 
- `PostID -> Content` 

Then what you do is the following: 

1. Iterate over all the `PostID -> Content` entries and read thei contents. 
2. Iterate over all the `PostID -> Index` entries and re-order the posts. 

What I was suggesting is changing one of the two relationships. The ones that I thought were: 

- `Index -> PostID` 
- `PostID -> Content` 

This way, since the `Index` will be an `sdk.Int`, if we iterate over all the `Index` keys they will be iterated over in an ascending way (from `1` to `n`). 

So the implementation should be something like

```golang
func (k Keeper) IteratePosts(ctx sdk.Context, fn func(index int64, post types.Post) (stop bool)) {
  store := ctx.KVStore(k.StoreKey)
  
  iterator := sdk.KVStorePrefixIterator(store, types.PostIndexStorePrefix)
  defer iterator.Close()
  
  i := int64(0)
  for ; iterator.Valid(); iterator.Next() {
    // iterator.Value() will be the post ID
    // We need to: 
    // 1. Read the post id
    // 2. Based on the post ID, get the content
  }
}
```

Reply:
sorry I forgot to push the last commit yesterday, you can now take a look at the implementation you've suggested